token-service/src/main/java/org/p2p/token/service/interactor/TokenServiceInteractor.kt

```diff
@@ -0,0 +1,41 @@
+package org.p2p.token.service.interactor
+
+import org.p2p.token.service.model.TokenServiceMetadata
+import org.p2p.token.service.model.TokenServiceNetwork
+import org.p2p.token.service.model.TokenServicePrice
+import org.p2p.token.service.repository.TokenServiceLocalRepository
+import org.p2p.token.service.repository.TokenServiceRemoteRepository
+
+internal class TokenServiceInteractor(
+    private val remoteRepository: TokenServiceRemoteRepository,
+    private val localRepository: TokenServiceLocalRepository
+) {
+
+    suspend fun loadPriceForTokens(chain: TokenServiceNetwork, tokenAddresses: List<String>) {
+        val result = remoteRepository.loadTokensPrice(chain = chain, addresses = tokenAddresses)
+        result.forEach { queryResult ->
+            localRepository.setTokensPrice(
```

Comment:
setTokensPrice -> saveTokensPrices
we are talking about local repository - so it saves smth inside